geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Transform.scala

```diff
@@ -0,0 +1,248 @@
+   * @return
+   */
+  def name: String
+
+  /**
+   * The type of the attribute
+   *
+   * @return
+   */
+  def binding: Class[_]
+
+  /**
+   * Evaluate the transform against a full feature
+   *
+   * @param feature simple feature, of the original, untransformed feature type
+   * @return
+   */
+  def evaluate(feature: SimpleFeature): AnyRef
+}
+
+/**
+  * Convenience methods for working with relational query projections (aka transforms)
+  */
+object Transform {
+
+  val DefinitionDelimiter = ";"
+
+  /**
+   * A simple relational projection (selection)
+   *
+   * @param name name
+   * @param binding binding
+   * @param i index of the original attribute being projected
+   */
+  case class PropertyTransform(name: String, binding: Class[_], i: Int) extends Transform {
+    override def evaluate(feature: SimpleFeature): AnyRef = feature.getAttribute(i)
+  }
+
+  /**
+   * A simple relational projection, but also renamed
+   *
+   * @param name name
+   * @param binding binding
+   * @param original name of the original attribute being projected
+   * @param i index of the original attribute being projected
+   */
+  case class RenameTransform(name: String, binding: Class[_], original: String, i: Int) extends Transform {
+    override def evaluate(feature: SimpleFeature): AnyRef = feature.getAttribute(i)
+  }
+
+  /**
+   * An attribute comprised of an expression (function, math transform, json-path, etc)
+   *
+   * @param name name
+   * @param binding binding
+   * @param expression expression
+   */
+  case class ExpressionTransform(name: String, binding: Class[_], expression: Expression) extends Transform {
+    override def evaluate(feature: SimpleFeature): AnyRef = expression.evaluate(feature)
+  }
+
+  object Transforms {
+
+    /**
+     * Create transform definitions for a delimited expression string
+     *
+     * @param sft original simple feature type
+     * @param transforms semi-colon delimited transform expressions
+     * @return
+     */
+    def apply(sft: SimpleFeatureType, transforms: String): Seq[Transform] =
+      if (transforms.isEmpty) { Seq.empty } else { apply(sft, transforms.split(DefinitionDelimiter)) }
+
+    /**
+     * Create transform definitions for a sequence of expression strings
+     *
+     * @param sft original simple feature type
+     * @param transforms transform expressions
+     * @return
+     */
+    def apply(sft: SimpleFeatureType, transforms: Seq[String]): Seq[Transform] =
+      transforms.map(definition(sft, _))
+
+    /**
+     * Create a transform definition from an expression
+     *
+     * @param sft simple feature type
+     * @param transform transform expression
+     * @return
+     */
+    private def definition(sft: SimpleFeatureType, transform: String): Transform = {
+      val equals = transform.indexOf('=')
+      if (equals == -1) {
+        val name = transform.trim()
+        val i = sft.indexOf(name)
+        if (i == -1) {
+          attributeExpression(sft, name)
```

Comment:
What's an example of a string that hits this branch?

Reply:
the main use case is json-path selectors like `$.foo.bar` - other cases would probably be xpath, or anything that uses a non-standard PropertyAccessor.